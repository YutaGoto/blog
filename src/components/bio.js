import * as React from "react"
import { StaticQuery, graphql } from "gatsby"
import { StaticImage } from "gatsby-plugin-image"
import "./fontawesome"

export const PureBio = ({ data }) => {
  const author = data.site.siteMetadata?.author
  const social = data.site.siteMetadata?.social

  return (
    <div className="px-4 py-10 bg-card dark:bg-card-dark shadow-lg sm:rounded-xl sm:p-5">
      <div className="flex space-x-4">
        <div className="flex-none">
          <StaticImage
            className="bio-avatar"
            layout="fixed"
            formats={["AUTO", "WEBP", "AVIF"]}
            src="../images/profile-pic.png"
            width={180}
            height={180}
            quality={95}
            alt="Profile picture"
          />
        </div>

        <div className="flex-1 flex flex-col space-y-3">
          <div>
            <p className="text-3xl mb-2">{author.name}</p>
            <p className="text-xl">{author.summary}</p>
          </div>
          <div className="flex-1 flex space-x-6">
            {social.map(s => (
              <a
                key={s.name}
                href={s.url}
                className="object-bottom"
                area-label={s.name}
                target="_blank"
                rel="noreferrer noopener"
              >
                <span className="icon">
                  <i
                    className={`fab fa-${s.name} fa-lg`}
                    aria-hidden="true"
                  ></i>
                </span>
              </a>
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}

export const Bio = () => (
  <StaticQuery
    query={graphql`
      query BioQuery {
        site {
          siteMetadata {
            author {
              name
              summary
            }
            social {
              name
              url
            }
          }
        }
      }
    `}
    render={data => <PureBio data={data} />}
  />
)
