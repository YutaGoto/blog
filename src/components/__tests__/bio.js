import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"
import { PureBio as Bio } from "../bio"

describe("bio", () => {
  const data = {
    site: {
      siteMetadata: {
        title: `.ごっちの日記`,
        author: {
          name: `Yuta Goto`,
          summary: `エンジニアです。`,
        },
        description: `.ごっちのつれづれなる日記`,
        siteUrl: `https://yutagoto.github.io/`,
        social: [
          {
            name: `twitter`,
            url: `https://twitter.com/gggooottto`,
          },
          {
            name: `github`,
            url: `https://github.com/YutaGoto`,
          },
        ],
      },
    },
  }

  it("renders correctly", () => {
    const tree = renderer.create(<Bio data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("display the nav element", () => {
    const { container } = render(<Bio data={data} />)
    expect(container.querySelector(`nav`)).toBeInTheDocument()
  })
})
