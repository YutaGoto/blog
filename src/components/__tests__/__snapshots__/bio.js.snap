// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bio renders correctly 1`] = `
<div
  className="px-4 py-10 bg-gray-50 dark:bg-gray-900 shadow-lg sm:rounded-xl sm:p-5"
>
  <div
    className="flex space-x-4"
  >
    <div
      className="flex-none"
    />
    <div
      className="flex-1"
    >
      <div>
        <p>
          <strong>
            Yuta Goto
          </strong>
          <br />
          エンジニアです。
        </p>
      </div>
      <div
        className="flex space-x-4"
      >
        <a
          area-label="twitter"
          className=""
          href="https://twitter.com/gggooottto"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            className="icon"
          >
            <i
              aria-hidden="true"
              className="fab fa-twitter"
            />
          </span>
        </a>
        <a
          area-label="github"
          className=""
          href="https://github.com/YutaGoto"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            className="icon"
          >
            <i
              aria-hidden="true"
              className="fab fa-github"
            />
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
